### PR TITLE
document environment variables and constants

### PR DESCRIPTION
**Pull request by cosmic-byte**
## Description
Document environment variables and constants

## TODO

Use this section for work-in-progress pull requests. If you're using this section,
please tag your PR "WIP". 
- [ ] Documentation

### CODE REVIEW

File: docs/API Reference/cli.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Quilt3 CLI
+# Quilt3 CLI and Shell
```
Comment (akarve) [inline docs/API Reference/cli.md:1]:
```suggestion
# Quilt3 CLI and environment
```

File: docs/API Reference/cli.md [hunk 2]
```diff
@@ -158,3 +158,24 @@ optional arguments:
   --dest DEST          Where to copy the objects in the package
   --message MESSAGE    The commit message for the new package
 ```
+## Environment variables
+#### `QUILT_DISABLE_USAGE_METRICS`
```
Comment (akarve) [inline docs/API Reference/cli.md:162]:
```suggestion
### `QUILT_DISABLE_USAGE_METRICS`
```

File: docs/API Reference/cli.md [hunk 3]
```diff
@@ -158,3 +158,24 @@ optional arguments:
...
+Disable anonymous usage collection. Defaults to `False`
+```
+$ export QUILT_DISABLE_USAGE_METRICS=true
+```
+#### `QUILT_MINIMIZE_STDOUT`
```
Comment (akarve) [inline docs/API Reference/cli.md:167]:
```suggestion
### `QUILT_MINIMIZE_STDOUT`
```

File: docs/API Reference/cli.md [hunk 4]
```diff
@@ -158,3 +158,24 @@ optional arguments:
...
+Turn off TQDM progress bars for log files. Defaults to `False`
+```
+$ export QUILT_MINIMIZE_STDOUT=true
+```
+## Constants
```
Comment (akarve) [inline docs/API Reference/cli.md:172]:
```suggestion
## Constants (see [util.py](https://github.com/quiltdata/quilt/blob/master/api/python/quilt3/util.py) for more) 
```

File: docs/API Reference/cli.md [hunk 5]
```diff
@@ -158,3 +158,24 @@ optional arguments:
...
+#### `APP_NAME`
```
Comment (akarve) [inline docs/API Reference/cli.md:173]: Make this a list instead and add a description
* `APP_NAME`, etc.

File: gendocs/gen_cli_api_reference.sh [hunk 6]
```diff
@@ -26,6 +38,28 @@ gen_cmd_docs 'disable-telemetry'
...
+gen_env_docs QUILT_DISABLE_USAGE_METRICS '$ export QUILT_DISABLE_USAGE_METRICS=true' \
+'Disable anonymous usage collection. Defaults to `False`'
+
+gen_env_docs QUILT_MINIMIZE_STDOUT '$ export QUILT_MINIMIZE_STDOUT=true' \
+'Turn off TQDM progress bars for log files. Defaults to `False`'
+
+
+# Document Constants
+echo '## Constants (see [util.py](https://github.com/quiltdata/quilt/blob/master/api/python/quilt3/util.py) for more)' >> cli.md
+echo '- `APP_NAME` - Name of the application' >> cli.md
```
Comment (akarve) [inline gendocs/gen_cli_api_reference.sh:52]:
```suggestion
echo '- `APP_NAME` >> cli.md
```

File: gendocs/gen_cli_api_reference.sh [hunk 7]
```diff
@@ -26,6 +38,28 @@ gen_cmd_docs 'disable-telemetry'
...
+echo '- `APP_AUTHOR` - Author' >> cli.md
```
Comment (akarve) [inline gendocs/gen_cli_api_reference.sh:53]:
```suggestion
echo '- `APP_AUTHOR` >> cli.md
```

File: gendocs/gen_cli_api_reference.sh [hunk 8]
```diff
@@ -26,6 +38,28 @@ gen_cmd_docs 'disable-telemetry'
...
+echo '- `BASE_DIR` - Base directory of the application' >> cli.md
+echo '- `BASE_PATH` - Base path(pathlib) of the application directory' >> cli.md
```
Comment (akarve) [inline gendocs/gen_cli_api_reference.sh:55]:
```suggestion
echo '- `BASE_PATH` - Base pathlib path for the application directory' >> cli.md
```

File: gendocs/gen_cli_api_reference.sh [hunk 9]
```diff
@@ -26,6 +38,28 @@ gen_cmd_docs 'disable-telemetry'
...
+echo '- `CACHE_PATH` - User cache directory ' >> cli.md
+echo '- `TEMPFILE_DIR_PATH` - Tempfile directory' >> cli.md
+echo '- `CONFIG_PATH` - Application configuration directory' >> cli.md
+echo '- `OPEN_DATA_URL` - Application data url' >> cli.md
+echo '- `PACKAGE_NAME_FORMAT` - Package name format' >> cli.md
```
Comment (akarve) [inline gendocs/gen_cli_api_reference.sh:60]:
```suggestion
echo '- `PACKAGE_NAME_FORMAT` - Regex for legal package names' >> cli.md
```

File: gendocs/gen_cli_api_reference.sh [hunk 10]
```diff
@@ -26,6 +38,28 @@ gen_cmd_docs 'disable-telemetry'
...
+
+
+# Document Constants
+echo '## Constants (see [util.py](https://github.com/quiltdata/quilt/blob/master/api/python/quilt3/util.py) for more)' >> cli.md
+echo '- `APP_NAME` - Name of the application' >> cli.md
+echo '- `APP_AUTHOR` - Author' >> cli.md
+echo '- `BASE_DIR` - Base directory of the application' >> cli.md
+echo '- `BASE_PATH` - Base path(pathlib) of the application directory' >> cli.md
+echo '- `CACHE_PATH` - User cache directory ' >> cli.md
+echo '- `TEMPFILE_DIR_PATH` - Tempfile directory' >> cli.md
```
Comment (akarve) [inline gendocs/gen_cli_api_reference.sh:57]: These comments should either be useful :) or delete them. You might have to read the code a bit do see what they do.

File: gendocs/gen_cli_api_reference.sh [hunk 11]
```diff
@@ -26,6 +38,28 @@ gen_cmd_docs 'disable-telemetry'
...
+
+gen_env_docs QUILT_MINIMIZE_STDOUT '$ export QUILT_MINIMIZE_STDOUT=true' \
+'Turn off TQDM progress bars for log files. Defaults to `False`'
+
+
+# Document Constants
+echo '## Constants (see [util.py](https://github.com/quiltdata/quilt/blob/master/api/python/quilt3/util.py) for more)' >> cli.md
+echo '- `APP_NAME`' >> cli.md
+echo '- `APP_AUTHOR`' >> cli.md
+echo '- `BASE_DIR` - Base directory of the application' >> cli.md
```
Comment (akarve) [inline gendocs/gen_cli_api_reference.sh:54]: is this not pathlib?

Comment (cosmic-byte) [reply gendocs/gen_cli_api_reference.sh:54]: No it's not.

Comment (cosmic-byte) [reply gendocs/gen_cli_api_reference.sh:54]: This is how it is defined.
```
BASE_DIR = user_data_dir(APP_NAME, APP_AUTHOR)
```
and `user_data_dir` returns a `os.path`

**Pull request merged**